packages/components/tree/src/tree.ts

```diff
@@ -0,0 +1,263 @@
+    'sl-tree': Tree;
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type TreeItemRenderer<T = any> = (item: TreeDataSourceNode<T>) => TemplateResult;
+
+/**
+ * A tree component. Use this if you have hierarchical data that you want
+ * to visualize.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class Tree<T = any> extends ScopedElementsMixin(LitElement) {
+  /** @internal */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-icon': Icon,
+      'sl-skeleton': Skeleton,
+      'sl-spinner': Spinner,
+      'sl-tree-node': TreeNode
+    };
+  }
+
+  /** @internal */
+  static override shadowRootOptions: ShadowRootInit = { ...LitElement.shadowRootOptions, delegatesFocus: true };
+
+  /** @internal */
+  static override styles: CSSResultGroup = styles;
+
+  /** The data model for the tree. */
+  #dataSource?: TreeDataSource<T>;
+
+  /** Manage keyboard navigation between tabs. */
+  #rovingTabindexController = new RovingTabindexController<TreeNode<T>>(this, {
+    focusInIndex: (elements: Array<TreeNode<T>>) => elements.findIndex(el => !el.disabled),
+    elements: () => Array.from(this.shadowRoot?.querySelectorAll('sl-tree-node') ?? []),
+    isFocusableElement: (el: TreeNode<T>) => !el.disabled
+  });
+
+  /** The virtualizer instance. */
+  #virtualizer?: VirtualizerHostElement[typeof virtualizerRef];
+
+  get dataSource() {
+    return this.#dataSource;
+  }
+
+  /** The model for the tree. */
+  @property({ attribute: false })
+  set dataSource(dataSource: TreeDataSource<T> | undefined) {
+    if (this.#dataSource) {
+      this.#dataSource.removeEventListener('sl-update', this.#onUpdate);
+    }
+
+    this.#dataSource = dataSource;
+    this.#dataSource?.addEventListener('sl-update', this.#onUpdate);
+    this.#dataSource?.update();
+  }
+
+  /** Hides the indentation guides when set. */
+  @property({ type: Boolean, attribute: 'hide-guides' }) hideGuides?: boolean;
+
+  /**
+   * Use this if you want to wait until lit-virtualizer has finished the rendering
+   * the tree nodes. This can be useful in unit tests for example.
+   */
+  get layoutComplete() {
+    return this.#virtualizer?.layoutComplete ?? Promise.resolve();
+  }
+
+  /** Custom renderer function for tree items. */
+  @property({ attribute: false }) renderer?: TreeItemRenderer<T>;
+
+  /**
+   * The custom elements used for rendering this tree. If you are using a custom renderer
+   * to render the tree nodes, any custom elements you use in the renderer need to be specified
+   * via this property. Otherwise those custom elements will not initialize, since the tree
+   * uses a Scoped Custom Element Registry.
+   */
+  @property({ attribute: false }) scopedElements?: Record<string, typeof HTMLElement>;
+
+  /** @internal Emits when the user selects a tree node. */
+  @event({ name: 'sl-select' }) selectEvent!: EventEmitter<SlSelectEvent<TreeDataSourceNode<T>>>;
```

Comment:
I'm just thinking, maybe we should rename SlSelectEvent from `sl-select` to sth else? Since `sl-select` is also used for select component?

Reply:
If we change it, let's do it in a different PR :)

Reply:
hmm maybe `sl-choose` or sth? Sounds a little bit weird, but still I think that maybe it would be better with a different name than `select` component. Maybe @Diaan do you have any other idea? :)